packages/flutter/lib/src/material/app_bar_theme.dart

```diff
@@ -5,111 +5,191 @@
 import 'dart:ui' show lerpDouble;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
 import 'text_theme.dart';
 import 'theme.dart';
 
-/// Defines default property values for descendant [AppBar] widgets.
+/// Overrides the default values of visual properties for descendant
+/// [AppBar] widgets.
 ///
-/// Descendant widgets obtain the current [AppBarTheme] object using
+/// Descendant widgets obtain the current [AppBarTheme] object with
 /// `AppBarTheme.of(context)`. Instances of [AppBarTheme] can be customized
 /// with [AppBarTheme.copyWith].
 ///
 /// Typically an [AppBarTheme] is specified as part of the overall [Theme] with
 /// [ThemeData.appBarTheme].
 ///
 /// All [AppBarTheme] properties are `null` by default. When null, the [AppBar]
-/// will use the values from [ThemeData] if they exist, otherwise it will
-/// provide its own defaults.
-///
-/// See also:
-///
-///  * [ThemeData], which describes the overall theme information for the
-///    application.
+/// compute its own default values, typically based on the overall theme's
+/// [ThemeData.colorScheme], [ThemeData.textTheme], and [ThemeData.iconTheme].
 @immutable
 class AppBarTheme with Diagnosticable {
   /// Creates a theme that can be used for [ThemeData.appBarTheme].
   const AppBarTheme({
     this.brightness,
     this.color,
+    this.backgroundColor,
+    this.foregroundColor,
     this.elevation,
     this.shadowColor,
     this.iconTheme,
     this.actionsIconTheme,
     this.textTheme,
     this.centerTitle,
     this.titleSpacing,
+    this.toolbarTextStyle,
+    this.titleTextStyle,
+    this.systemOverlayStyle,
   });
 
-  /// Default value for [AppBar.brightness].
+  /// This property is obsolete, please use [systemOverlayStyle] instead.
```

Comment:
Should we mark this as deprecated, or are you waiting to get things migrated first?